view/frontend/web/js/view/suggested_address_checkout_step.js

```diff
@@ -106,15 +114,34 @@ function (
             var addrs = avCore.suggestedAddresses();
 
             if (addrs && addrs.length) {
-                var newAddr = $.extend({}, quote.shippingAddress(), addrs[id].address, { custom_attributes: { suggestedAddress: true } });
+                var checkoutProvider = registry.get('checkoutProvider');
+                var originalAddress = $.extend({}, checkoutProvider.get('shippingAddress'));
+
+                originalAddress.city = addrs[id].address.city;
+                originalAddress.country_id = addrs[id].address.countryId;
+                originalAddress.postcode = addrs[id].address.postcode;
+                originalAddress.region_id = addrs[id].address.regionId;
 
-                // Force shipping rates to recalculate
-                // https://alanstorm.com/refresh-shipping-rates-for-the-magento-2-checkout/
-                rateRegistry.set(newAddr.getKey(), null);
-                rateRegistry.set(newAddr.getCacheKey(), null);
+                addrs[id].address.street.forEach(function(item, index) {
+                   originalAddress.street[index] = item;
+                });
 
-                quote.shippingAddress(newAddr);
-                quote.billingAddress(newAddr);
+                if (id !== 0) {
+                    originalAddress.taxjar_attributes = {
+                        suggestedAddress: true
+                    };
+                } else {
+                    originalAddress.taxjar_attributes = {
+                        suggestedAddress: false
+                    };
+                }
+
+                // street address does not update when updating the checkout provider shipping address
+                // that input must be updated individually
+                $('.form-shipping-address input[name="street[0]"]').val(originalAddress.street[0]);
```

Comment:
Feels dirty to directly reach into the form and update it. Is this a typical thing to do?  Is there a way to do this via Magento? (i'm not familiar with this generally works).

Reply:
I don't particularly like this approach either. However when updating through the "normal" magento way (the checkoutProvider.set line) it does not update the street address. None of the other methods I tested would correctly update the street address to I had to revert to the approach in this PR.